cpp/engine.cpp

```diff
@@ -170,9 +170,16 @@ Engine::Engine(util::Dir *data_dir, const char *windowtitle)
 	if (devices.empty()) {
 		throw util::Error{MSG(err) << "No audio devices found"};
 	}
+
+	// initialize profiler
+	this->profiler.register_category(util::Profiler::Categorie::EVENT_PROCESSING);
+	this->profiler.register_category(util::Profiler::Categorie::RENDERING);
+	this->profiler.register_category(util::Profiler::Categorie::IDLE_TIME);
```

Comment:
It would be best if we could avoid havoing to manually call this for every member of the enum (what happens if I forget to do it? Undefined behavior?)